src/components/icon/js/iconService.js

```diff
@@ -422,8 +422,25 @@
     */
    function cacheIcon( id ) {
 
-     return function updateCache( icon ) {
-       iconCache[id] = isIcon(icon) ? icon : new Icon(icon, config[id]);
+     return function updateCache( _icon ) {
+       var icon = isIcon(_icon) ? _icon : new Icon(_icon, config[id]);
+
+       //clear id attributes to prevent aria issues
+       var elem = icon.element;
+       elem.removeAttribute('id');
+
+       var children = elem.childNodes;
+
+       for (var i=0; i<children.length;i++) {
+         var child = children[i];
+         if(child.nodeName==='path') {
```

Comment:
@david-gang I suggest, walking recursively through the children and remove the `id` if present.
Like so
```js
(function recursiveIteration(element) {
  angular.forEach(element.childNodes, function(item) {
    item.removeAttribute('id');
    if (item.childNodes) {
      recursiveIteration(item)
    }
  });
})(myIconElement);
```

Or another way, would be the `querySelector`. Like so
```js
angular.forEach(myIconElement.querySelectorAll('[id]'), function(item) {
  item.removeAttribute('id');
});
```

But I'm not sure what's the most elegant / performant solution.